spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -546,8 +565,11 @@ case class DeltaSharingSource(
 
     val fileActions = sortedFetchedFiles.takeWhile {
       case IndexedFile(version, index, _, _, _, _) =>
-        version < endOffset.tableVersion ||
-          (version == endOffset.tableVersion && index <= endOffset.index)
+        // Ensure (version, index) is in the range of
+        // [(startVersion, startIndex), (endVersion, endOffset)]
+        (version > startVersion || (version == startVersion && (index == -1 ||
+          index >= startIndex))) && (version < endOffset.tableVersion ||
+          (version == endOffset.tableVersion && index <= endOffset.index))
```

Comment:
Question: Can these changes be unit tested?
Could we call the DeltaSharingSource apis from a test that can mimic various scenarios?

Reply:
It's hard to test because spark doesn't allow `getBatch` to be called alone: https://github.com/delta-io/delta-sharing/blob/main/spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala#L170

But I'll run integration tests in the notebook. I'll ping you specific tests before merging it.